doc/features/metrics/README.md

```diff
@@ -0,0 +1,119 @@
+|-----------------------------------|-------------------------|----------------------------------------|
+| `prefix.s0`                       | `connected-nodes`       | `SessionMetric.Gauges.ConnectedNodes`  |
+| `prefix.s0.nodes.127_0_0_1:9042`  | `pool.open-connections` | `NodeMetric.Gauges.OpenConnections`    |
+
+## Configuration
+
+By default, if `Builder.WithMetrics` is not called by the client application, metrics are disabled.
+
+If `Builder.WithMetrics(provider)` is called, the default metrics will be enabled. The default metrics are every metrics except those of type `Timer`. The reasoning behind this is the fact that enabling `Timer` metrics have the potential effect of reducing the throughput of the driver so it's recommended to benchmark the client application with `Timer` metrics enabled before enabling them in production.
+
+`Builder.WithMetrics(IDriverMetricsProvider,DriverMetricsOptions)` can be used to customize options related to metrics. `DriverMetricsOptions.SetEnabledNodeMetrics()` and `DriverMetricsOptions.SetEnabledSessionMetrics()` can be used to specify which metrics should be enabled. **Note that this will override the default enabled metrics**, i.e., if you want to enable `Timer` metrics on top of the default metrics then all metrics must be specified. `SessionMetric` and `NodeMetric` have a couple of `static` properties to make it easier to specify which metrics to enable. Here are some examples:
+
+```csharp
+// Enable all session metrics
+builder.WithMetrics(provider, new DriverMetricsOptions().SetEnabledSessionMetrics(SessionMetric.AllSessionMetrics));
+
+// Enable default session metrics (same effect as not calling the method at all)
+builder.WithMetrics(provider, new DriverMetricsOptions().SetEnabledSessionMetrics(SessionMetric.DefaultSessionMetrics));
+
+// Enable default session metrics except "bytes-sent", using LINQ
+builder.WithMetrics(provider, new DriverMetricsOptions()
+    .SetEnabledSessionMetrics(SessionMetric.DefaultSessionMetrics.Except(new [] { SessionMetric.Meters.BytesSent })));
+
+// Enable default session metrics and 'cql-requests' timer metric, using LINQ
+builder.WithMetrics(provider, new DriverMetricsOptions()
+    .SetEnabledSessionMetrics(SessionMetric.DefaultSessionMetrics.Union(new [] { SessionMetric.Timers.CqlRequests })));
+```
+
+`DriverMetricsOptions.SetBucketPrefix` can be used to specify a custom prefix to be added to the bucket name of every metric. See the [previous section](#Structure) for information about his.
+
+The `provider` parameter must be an implementation of `IDriverMetricsProvider`. As was mentioned previouly, DataStax offers an [App.Metrics based implementation on a separate  extension nuget package].
+
+## Retrieving metrics
+
+The driver's API for retrieving metrics is mostly an abstraction. The available methods on each metric type will depend on the provider that is plugged in to the driver.
```

Comment:
We could recommend to use built-in reporters in the metrics provider to output metrics information (like Graphite/Grafana), noting that "additionally" we expose an API to retrieve it.

Reply:
done